src/west/manifest.py

```diff
@@ -892,8 +890,67 @@ def _import_content_from_project(self, project, path):
 
         return content
 
-    def _import_map_from_project(self, project, map, projects):  # TODO
-        raise NotImplementedError('import: <map> from project unimplemented')
+    def _import_map(self, p, imp, base_importer, projects):
+        # Helper routine used to handle imports from self and projects.
+        # We import everything, then filter out what's not desired.
+
+        imap = self._load_imap(p, imp)
+
+        all_imported = {}
+        base_importer(p, imap.file, all_imported)
+
+        for name, project in all_imported.items():
+            if _is_imap_ok(project, imap):
+                if name in imap.rename:
+                    project.name = imap.rename[name]
```

Comment:
Seems you're only doing a rename of the project name.
What would happen if `path` had been defined in the project, and this path is colliding with the path of a similar project in downstream `west.yml` ?

Reply:
I'm not sure I understand your question. Renaming is meant to catch cases where I have something called 'app' both upstream and downstream, and I want to have both without having to track the upstream version in my west.yml. I don't see how a path collision is possible, since project.path is not changed and we already check for project path collisions by calling _check_paths_are_unique() from _load().

Reply:
Perhaps we should remove the 'rename' feature to get started, since it doesn't make as much sense now that we've moved to your suggested 'ignore duplicate definitions' semantics?

Looking at it again, it has the feature of something that hasn't been thought through enough yet. It is easy to put back if we get users asking for it.